### ISSUE TITLE
(DB/Quest): Dissension Amongst the Ranks...

### ISSUE THREAD

**Issue by TheWatchingI** [azerothcore/azerothcore-wotlk#3009]
##### SMALL DESCRIPTION:
The quest item doesn't work.

##### EXPECTED BLIZZLIKE BEHAVIOUR:
When equiping the tabard/the quest item, you're supposed to turn into a bloodelf.

##### CURRENT BEHAVIOUR:
The tabard has no effect upon equiping and you can't continue the quest. https://www.wowhead.com/item=31279/enchanted-illidari-tabard

##### STEPS TO REPRODUCE THE PROBLEM:
1. Get the quest https://www.wowhead.com/quest=10776/dissension-amongst-the-ranks
2. Equip the tabard provided by the quest giver https://www.wowhead.com/item=31279/enchanted-illidari-tabard
3. Kill a Crazed Colussus, if you want to test, if it counts toward the quest completion https://www.wowhead.com/npc=19823/crazed-colossus

##### EXTRA NOTES:
The quest exists two times: One for the Alliance and one for the Horde. I only tested the Alliance one.

##### BRANCH(ES):
master

##### AC HASH/COMMIT:
cdf518a

##### OPERATING SYSTEM:
Windows 10

##### MODULES:
ah-bot
morphsummon
solo-lfg

**Comment by Kitzunu** [comment]
The disguise works.
![image](https://user-images.githubusercontent.com/24550914/81440516-7e53b080-9170-11ea-8907-7c691c9a1738.png)

> "The illusion effect of this item only works at the Eclipsion Fields."

But I will try to actually fix the quest

**Comment by Kitzunu** [comment]
honestly I cannot figure out why 
```
ID - 38223 Quest Credit: Crazed Colossus
=================================================
Category = 0, SpellIconID = 1, activeIconID = 0, SpellVisual = (0,0), SpellPriority = 0
Family SPELLFAMILY_GENERIC, flag [0] 0x00000000 [1] 0x00000000 [2] 0x00000000

SpellSchoolMask = 1 (SPELL_SCHOOL_MASK_NORMAL)
DamageClass = 0 (SPELL_DAMAGE_CLASS_NONE)
PreventionType = 0 (SPELL_PREVENTION_TYPE_NONE)
=================================================
Attributes: 0x00800000 (SPELL_ATTR0_CASTABLE_WHILE_DEAD)
AttributesEx2: 0x00000004 (SPELL_ATTR2_CAN_TARGET_NOT_IN_LOS)
=================================================
Spell Level = 0, base 0, max 0, maxTarget 0

Category = 0
DispelType = 0 (DISPEL_NONE)
Mechanic = 0 (MECHANIC_NONE)
SpellRange: (Id 13) "Anywhere":
    MinRange = 0, MinRangeFriendly = 0
    MaxRange = 50000, MaxRangeFriendly = 50000

CastingTime (Id 1) = 0,00

Interrupt Flags: 0x00000000, AuraIF 0x00000000, ChannelIF 0x00000000
Chance = 101, charges - 0
=================================================
Effect 0: Id 77 (SPELL_EFFECT_SCRIPT_EFFECT)
BasePoints = 38224
Targets (25, 37) (TARGET_UNIT_TARGET_ANY, TARGET_UNIT_LASTTARGET_AREA_PARTY)

Effect 1:  NO EFFECT

Effect 2:  NO EFFECT
```
is not giving the quest credit... I have to investigate this another time

**Comment by TheWatchingI** [comment]
> "The illusion effect of this item only works at the Eclipsion Fields."

I'm so incredible sorry! I fought the quest mob and thought that means I'm in the right location, but I was only beside it. I'm really sorry, you had to put work into something, which isn't a bug. The quest works as intended.

**Comment by meeko-dev** [comment]
This issue still exists. It's not only the illusion which works only if you equip the tabard in the correct zone. The aura, which is needed for kill credit (ID 38224) also gets applied only if you equip the tabard in Eclipsion Fields (area ID 3520). What's weird is that the aura gets removed upon leaving the zone but it doesn't get reapplied upon entering the zone again while wearing the tabard.

The `Register` function in the AuraScript only runs when the tabard is equipped in the correct area and it attaches `OnEffectApply` and `OnEffectRemove` listeners, however the `OnEffectApply` never runs upon entering the zone while the `OnEffectRemove` runs upon leaving it.

Also what's even weirder is that if you shapeshift in or out of a druid form while wearing the tabard and stading in the correct area, the `Register` function runs again, therefore the `OnEffectApply` listener gets triggered and it applies the illusion aura and also the one needed for the kill credit.

So basically it looks like shapeshifting retriggers the on-equip effect of the tabard, while entering the zone doesn't trigger it, while also leaving the zone triggers its removal.
